File: backend/src/main/scala/microservices/traveler/objects/TravelerDeletedPlannerResponse.scala
// 本文件定义 traveler 模块的删除响应对象 `TravelerDeletedPlannerResponse`，仅负责数据结构与 JSON codec。
package com.typesafe.travel.traveler.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class TravelerDeletedPlannerResponse(
    deleted: Boolean,
    hidden: Boolean
)

object TravelerDeletedPlannerResponse:
  given sourceEncoder: Encoder[TravelerDeletedPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[TravelerDeletedPlannerResponse] = deriveDecoder

File: frontend/src/microservices/traveler/objects/TravelerDeletedPlannerResponse.ts
// 本文件定义 traveler 模块的 `TravelerDeletedPlannerResponse`，作为删除响应数据并提供 JSON 编解码。

export type TravelerDeletedPlannerResponse = {
  deleted: boolean
  hidden: boolean
}

export const travelerDeletedPlannerResponseFromJson = (json: string): TravelerDeletedPlannerResponse =>
  JSON.parse(json) as TravelerDeletedPlannerResponse

export const travelerDeletedPlannerResponseToJson = (value: TravelerDeletedPlannerResponse): string =>
  JSON.stringify(value)
